indexer/controller/coalesce.go

```diff
@@ -11,75 +11,79 @@ import (
 	"github.com/quay/claircore/indexer"
 )
 
-// coalesce calls each ecosystem's coalescer and merges the returned IndexReports
+// Coalesce calls each ecosystem's Coalescer and merges the returned IndexReports.
 func coalesce(ctx context.Context, s *Controller) (State, error) {
-	cctx, cancel := context.WithCancel(ctx)
-	defer cancel()
 	mu := sync.Mutex{}
 	reports := []*claircore.IndexReport{}
-	g := errgroup.Group{}
-	// dispatch a coalescer go routine for each ecosystem
+	g, gctx := errgroup.WithContext(ctx)
+	// Dispatch a Coalescer goroutine for each ecosystem.
 	for _, ecosystem := range s.Ecosystems {
+		select {
+		case <-gctx.Done():
+			break
+		default:
+		}
 		artifacts := []*indexer.LayerArtifacts{}
-		pkgScanners, _ := ecosystem.PackageScanners(cctx)
-		distScanners, _ := ecosystem.DistributionScanners(cctx)
-		repoScanners, _ := ecosystem.RepositoryScanners(cctx)
+		pkgScanners, _ := ecosystem.PackageScanners(gctx)
```

Comment:
I guess the idea behind sharing this context is to be able to early out if the Coalesce goroutine fails and execution has already passed the `gctx.Done()` check?